src/index.ts

```diff
@@ -70,6 +76,16 @@ class Client {
             throw new Error('A port is required');
         }
 
+        this.debug = debuglog('dats');
+
+        if (debug) {
```

Comment:
The debug util already has the logic to know if it's enabled or not 🤔 . I think we could avoid all this checks and just use the debug util as it is, without allowing overriding. We would treat it as an internal.